### PR TITLE
Make cse variables parameters that are added to globalKnownVars

### PR DESCRIPTION
**Pull request by lochel**
This is part of a more complex debugging session with @AnHeuermann, @kabdelhak and @arun3688. Some variables get wrong `fixed` attributes which has all sort of side-effects. This is testing some of the changes isolated from everything else we did.

### CODE REVIEW

File: OMCompiler/Compiler/BackEnd/BackendVariable.mo [hunk 1]
```diff
@@ -1248,6 +1248,13 @@ algorithm
   end match;
 end isParam;
 
+public function makeParam
+  "Change variable to parameter"
+  input output BackendDAE.Var var;
+algorithm
+  var.varKind := BackendDAE.PARAM();
```
Comment (kabdelhak) [inline OMCompiler/Compiler/BackEnd/BackendVariable.mo:1255]: Maybe restrict this to not overwrite `STATE()` and `STATE_DER()`? Fail for safety reasons?

File: OMCompiler/Compiler/BackEnd/CommonSubExpression.mo [hunk 2]
```diff
@@ -545,7 +545,7 @@ algorithm
 
             // Save the rhs (call) as bind expression and set fixed=true
             var := BackendVariable.setBindExp(var, SOME(call));
-            var := BackendVariable.setVarFixed(var, true);
+            var := BackendVariable.makeParam(var);
```
Comment (kabdelhak) [inline OMCompiler/Compiler/BackEnd/CommonSubExpression.mo:548]: You could also just inline the function call here, don't know if it is really necessary.

**Review by kabdelhak — APPROVED** [review]
LGTM

This looks like the correct approach. Some small change requests that are not essential.

File: OMCompiler/Compiler/BackEnd/BackendVariable.mo (hunk 1 above)
Comment (lochel) [reply OMCompiler/Compiler/BackEnd/BackendVariable.mo:1255]: I don't think that either of them can end up in the globalKnownVars. I will try to make an example to proof myself wrong.

File: testsuite/simulation/modelica/commonSubExp/wrapFunctionCalls12.mos [hunk 3]
```diff
@@ -34,9 +34,9 @@ simulate(CSE.wrapFunctionCalls12); getErrorString();
 //
 // ########### Updated globalKnownVars (simulation) (3)
 // ========================================
-// 1: x:VARIABLE(unit = "s" fixed = true )  = sin(p1)  type: Real 
-// 2: p1:PARAM()  = 5.0  type: Real 
-// 3: y:VARIABLE(unit = "m" fixed = true )  = cos(x)  type: Real 
+// 1: x:VARIABLE(unit = "s" fixed = true )  = sin(p1)  type: Real
```
Comment (phannebohm) [inline testsuite/simulation/modelica/commonSubExp/wrapFunctionCalls12.mos:37]: That's weird: if `y` is made `PARAM`, why isn't `x`?
Is this done before matching-sorting?

**Pull request merged**